### PR TITLE
Decision Tree: Update defaults for max_depth, num_splits

### PR DESCRIPTION
**Pull request by iyerr3**
Reduce the defaults for max_depth to 7 and num_splits to 20 to ensure we
don't run out of memory when initializing tree for problems with many
features or with features with many categorical values.

### CODE REVIEW

**Comment by ivannovick** [comment]
does this improve memory usage?

**Comment by iyerr3** [comment]
No, that's a separate JIRA: MADLIB-1057
<https://issues.apache.org/jira/browse/MADLIB-1057>. This one is just about
setting the defaults to a more reasonable value considering the data that
users have shared.

The commit is a little more than just changing two numbers since I updated
the way these defaults are set. Previously they were set in overloaded
function declaration (in SQL). Changed this to set the default in the main
function definition, eliminating redundancy.

Thanks,
Rahul

**Comment by iyerr3** [comment]
Reading the commit description again, I would rephrase it as 

"Reduce the defaults for max_depth to 7 and num_splits to 20 to **minimize
the chances of running out of memory** when initializing tree for problems 
with many features or with features with many categorical values."

**Comment by ivannovick** [comment]
How do we update the docs in synch with this change?
https://madlib.incubator.apache.org/docs/latest/group__grp__decision__tree.html

Is it needed?

**Comment by iyerr3** [comment]
The docs/latest corresponds to latest release (1.10) and won't be updated till the next release. We also have [docs/master](http://madlib.incubator.apache.org/docs/master/), which can reflect these changes. We haven't updated those since the 1.10 release - I can update if you're looking for the changes reflected online.

**Comment by fmcquillan99** [comment]
Ivan, the docs get updated when the release GA's.
